### ISSUE TITLE
unable to "shake off" a grab

### ISSUE THREAD

**Issue by rrrepos** [wmurphyrd/aframe-super-hands-component#158]
I am testing the super-hands component to move an object. This is on a desktop without a headset. For the same I am setting the grabbable gesture component. What I am noticing is that once I have moved the object I am unable to "shake-off" the moving viz. it continues to be in a state of grabbable. The code is simple

````
<a-box ... grabbable></a-box>
..
<a-entity raycaster="objects: .clickable" cursor="rayOrigin:mouse" super-hands="colliderEvent: raycaster-intersection; colliderEventProperty: els; colliderEndEvent: raycaster-intersection-cleared; colliderEndEventProperty: el"></a-entity>
````

Does something need to be set to disable the grabbable state?

Thanks

**Comment by wmurphyrd** [comment]
The default setting for grabbable is that a mouseup event terminates the grab. Do you experience the same issue in [this example](https://wmurphyrd.github.io/aframe-super-hands-component/examples/mouse/index-mouse.html)?

**Comment by rrrepos** [comment]
Thanks @wmurphyrd,

No. I attempted to incorporate it to my use-case encouraged by the above example. The main difference being instead of using `event-set` as you have in the example, I am using a component set to the entity. Currently, the component just does this:

```
AFRAME.registerComponent( 'move', {
    init: function() {
    this.el.addEventListener ( 'grab-start', function(evt) {
      console.log('grab-start');
      evt.preventDefault()
    });
    this.el.addEventListener ( 'grab-end', function(evt) {
      console.log('grab-end');
      evt.preventDefault()
    });
  }
});
```
I chose to build a component because I may require better programming control to effect a few changes while moving as well as at the end of the moving. One of the needs is that I would like to restrict the move to within some set parameters. But all that later once I have crossed the "shaking off" hurdle

**Comment by rrrepos** [comment]
Kindly check this glitch:https://aframe-superhands-test.glitch.me/

This is the behaviour.

a. I can move around using keyboard
b. I can turn around using mouse movement without drag because pointerLockeEnabled has been set to true.
c. I can click-and-move the box to a new position (as desired)
d. Release stops the movement (as desired)
e. Click and drag anywhere else on the scene and the box continues to move (not desired).

Where am I going wrong? Thanks

**Comment by wmurphyrd** [comment]
Ah it's clear now with the code. In the super-hands ecosystem, calling `preventDefault` on a gesture event is indicating that the gesture event has been consumed, so it will be ignored by other reaction components on an entity or its ancestors. So, in your `move` component, you are consuming the `'grab-end'` event, and therefore grabbable stays grabbing waiting for another `'grab-end'` that it can use. 

Canceling events is vital when you're crafting custom reaction components, but, for piggy-backing off of other reactions, the best method is to listen for the `'stateadded'`/`'stateremoved'` events, in this case where `event.detail === 'grabbed'`. This way your additional logic is only triggered when a grab has succeeded, whereas listening on the gesture events occurs whenever a grab is attempted.

**Comment by rrrepos** [comment]
Thanks @wmurphyrd : I tried a number of changes based on your comments. For some reason it is not working. 

I tried remove the event listener.

I tried adding state but for some reason shows null.

Perhaps it is my Monday blues. Will try to get the understanding of `states` and `events` again.

**Comment by gabrieljbaker** [comment]
Curious about a point mentioned above, @wmurphyrd , is there a way to get super hands to work with cursor=rayOrigin:mouse?

**Comment by wmurphyrd** [comment]
@gabrieljbaker yes check out https://wmurphyrd.github.io/aframe-super-hands-component/examples/mouse/index-mouse.html

@rrrepos I just tried the glitch and it seems to be working as far as I can tell. Is this not the behavior you're looking for: https://youtu.be/XvxVMaix_8w

**Comment by rrrepos** [comment]
@wmurphyrd : Thanks for the video. I think what you have not tried is this sequence (I just did)

1. Let cursor be outside of the box.
2. Take cursor to box.
3. Drag the box (it moves).
4. End the drag (stops moving).
5. Move cursor outside of the box (box does not move).
6. Now, drag the cursor when outside of the box and you will find the box moving. (not desirable)

Thanks. RR

**Comment by wmurphyrd** [comment]
Aw geez I should have seen this right away. Sorry. The config you have for super-hands doesn't match the [current raycaster component](https://aframe.io/docs/0.8.0/components/raycaster.html#events_raycaster_intersection_cleared). `colliderEndEventProperty` should be `clearedEls` not `el`. See https://glitch.com/edit/#!/ahead-crime for a working example. Did you find that config somewhere in my docs? Please let me know where so I can update it.

**Comment by rrrepos** [comment]
Thanks @wmurphyrd . Now it works. I found the el code in your examples:

https://wmurphyrd.github.io/aframe-super-hands-component/examples/#mouse

![screen shot 2018-09-27 at 8 36 19 am](https://user-images.githubusercontent.com/14859054/46121214-70b61500-c230-11e8-8cb5-262666c963d3.png)

**Comment by rrrepos** [comment]
Extremely kind of you to help. Now I can work on extending the functionality to restrict movement within some set parameters in the x and z direction. Possibly will create a custom component with a tick function for that. Will think up of something. Thanks.
